src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -12,11 +13,56 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     public class SecurityTokenDescriptor
     {
+        // TODO: At next major version (8.0), remove Audience and logic for combining with Audiences.
+        private HashSet<string> _audiences;
+
         /// <summary>
-        /// Gets or sets the value of the 'audience' claim.
+        /// Gets or sets the value of the 'audience' claim. Will be deprecated in favor of <see cref="Audiences"/> in the next
+        /// major version (8.x).
         /// </summary>
         public string Audience { get; set; }
 
+        /// <summary>
+        /// Gets or sets one or more audiences to include in the token's 'Aud' claim. Automatically removes duplicates and empty,
+        /// null, or whitespace-only strings. Does not use a threadsafe collection.
+        /// </summary>
+        public IEnumerable<string> Audiences {
+            get
+            {
+                // If Audiences isn't set, return null since this will be the behavior once Audience is removed.
+                if (_audiences.IsNullOrEmpty())
+                    return null;
+
+                // If both Audience and Audiences are set, return the union of the two.
+                else if (!string.IsNullOrEmpty(Audience))
```

Comment:
This could lead to a race condition, depending on if the Audience property is set before or after get is called?
Not a good idea to automatically combine properties.

Reply:
Agreed, that is why I commented it wasn't threadsafe. I initially had it separate but switched to merging due to the [earlier feedback here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2575#discussion_r1585725216). Any thoughts given that?

Reply:
Synced offline and decided to stop merging the audience and audiences properties.